app/classifier/tasks/drawing/markings-renderer.cjsx

```diff
@@ -65,7 +64,10 @@ module.exports = React.createClass
                 onDestroy: @handleDestroy.bind this, annotation, mark
 
               ToolComponent = drawingTools[toolDescription.type]
-              <ToolComponent key={mark._key} {...toolProps} {...toolEnv} {...toolMethods} />}
+
+              if i < @props.hideMarksBeforeIndex then null # hide previous marks
```

Comment:
This `if`/`else` formatting is pretty weird. Consider breaking these lines and indenting.